youtube_dl/extractor/imgur.py

```diff
@@ -102,23 +102,32 @@ def _real_extract(self, url):
 class ImgurAlbumIE(InfoExtractor):
     _VALID_URL = r'https?://(?:i\.)?imgur\.com/gallery/(?P<id>[a-zA-Z0-9]+)'
 
-    _TEST = {
+    _TESTS = [{
         'url': 'http://imgur.com/gallery/Q95ko',
         'info_dict': {
             'id': 'Q95ko',
         },
         'playlist_count': 25,
-    }
+    },{
+        'url': 'https://imgur.com/gallery/YcAQlkx',
+        'info_dict': {
+            'id': 'YcAQlkx',
+        },
+        'playlist_count': 1,
+    }]
 
     def _real_extract(self, url):
         album_id = self._match_id(url)
 
-        album_images = self._download_json(
-            'http://imgur.com/gallery/%s/album_images/hit.json?all=true' % album_id,
-            album_id)['data']['images']
+        album_img_data = self._download_json(
+            'http://imgur.com/gallery/%s/album_images/hit.json?all=true' % album_id,album_id)['data']
 
-        entries = [
-            self.url_result('http://imgur.com/%s' % image['hash'])
-            for image in album_images if image.get('hash')]
+        if(len(album_img_data) == 0):
+            entries = [self.url_result('http://imgur.com/%s' % album_id)]
```

Comment:
Agreed. But obviously `entries = self.url_result('http://imgur.com/%s' % album_id)` will not work.
Thought about this previously, and intentionally kept the list so that code does not break.
If they are to be removed, will need to do a bit of refactoring. Was thinking about some ideas, but could not come up anything neat.
Can you suggest something?